File: src/types/skill.ts
export type SkillApps = Record<string, boolean>;

export interface Skill {
    id: string;
    name: string;
    description: string;
    version?: string;
    contentPath: string;
    source: 'user' | 'project' | 'plugin';
    enabled: boolean;
    apps?: SkillApps;
}

File: src-tauri/src/models/skill.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

pub type SkillApps = HashMap<String, bool>;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Skill {
    pub name: String,
    pub content: String,
    pub file_path: String,
    pub source: SkillSource,
    #[serde(default)]
    pub apps: SkillApps,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum SkillSource {
    User,
    Project,
}
